controllers/component/predicates.go

```diff
@@ -5,8 +5,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
+// ComponentCreatedPredicate returns a predicate which filters out
+// only components that have are the result of a create event
+func ComponentCreatedPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return true
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			return true
```

Comment:
While this doesn't hurt things currently, might be better to set it to `false` for clarity in the future.